libgit/autogit_node_wrappers.go

```diff
@@ -275,13 +282,53 @@ func (tn tlfNode) ShouldCreateMissedLookup(
 		return false, ctx, libkbfs.File, ""
 	}
 
-	ctx = context.WithValue(ctx, ctxReadWriteKey, 1)
+	ctx = context.WithValue(ctx, ctxReadWriteKey, struct{}{})
 	if name != normalizedRepoName {
 		return true, ctx, libkbfs.Sym, normalizedRepoName
 	}
 	return true, ctx, libkbfs.Dir, ""
 }
 
+// RemoveDir implements the Node interface for tlfNode.
+func (tn tlfNode) RemoveDir(ctx context.Context, name string) (
+	removeHandled bool, err error) {
+	if ctx.Value(ctxSkipRemoveDirKey) != nil {
+		return false, nil
+	}
+
+	// Is this a legit repo?
+	_, _, err = GetRepoAndID(ctx, tn.am.config, tn.h, name, "")
+	if err != nil {
+		return false, err
+	}
+
+	if normalizeRepoName(name) != name {
+		// Can't remove a symlink as if it were a directory.
+		return false, nil
+	}
+	ctx = context.WithValue(ctx, ctxSkipRemoveDirKey, struct{}{})
```

Comment:
Could you explain how `ctxSkipRemoveDirKey` is used and why it's needed here? It seems `tn.am.Delete` only utilizes timeout/cancellation on the `ctx`, and I don't see a recursive call into this function. So in what cases can the check on this key at the beginning return?

Reply:
Oops, you're right, this isn't actually needed.  I'll remove it.  Thanks!